addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/private_id_disclosure.js

```diff
@@ -0,0 +1,90 @@
+// This script scans websocket messages for private IP V4 addresses as well as Amazon EC2 private hostnames
+// Private IP V$ such as: 10.x.x.x, 172.x.x.x, 192.168.x.x
+// Amazon EC2 private hostname: such as ip-10-0-56-78
+
+// * This Passive Scan base on http passive scan plugin:
+// ** org.zaproxy.zap.extension.pscanrules.TestInfoPrivateAddressDisclosure
+
+// * Regex test: https://regex101.com/r/SztaPj/2/
+
+// Author: Manos Kirtas (manolis.kirt@gmail.com)
+
+OPCODE_TEXT = 0x1;
+RISK_LOW 	= 1;
+CONFIDENCE_MEDIUM = 2;
+
+REGULAR_IP_OCTET = "(25[0-5]|2[0-4][0-9]|1?[0-9]{1,2})";
+REGULAR_PORTS = "(:(0|[1-9]\\d{0,3}|[1-5]\\d{4}|6[0-4]\\d{3}|65([0-4]\\d{2}|5[0-2]\\d|53[0-5]))\\b)?";
+
+var patternPre = [];
+
+/** Pattern for private IP V4 addresses as well as Amazon EC2 private hostnames */
+patternPre.push(
+    "(",
+    "\\b10\\.(",
+    REGULAR_IP_OCTET,
+    "\\.){2}",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\b172\\.",
+    "(3[01]|2[0-9]|1[6-9])\\.",
+    REGULAR_IP_OCTET,
+    "\\.",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\b192\\.168\\.",
+    REGULAR_IP_OCTET,
+    "\\.",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    // find IPs from AWS hostnames such as "ip-10-2-3-200"
+    "\\bip-10-(",
+    REGULAR_IP_OCTET,
+    "-){2}",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\bip-172-",
+    "(3[01]|2[0-9]|1[6-9])-",
+    REGULAR_IP_OCTET,
+    "-",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\bip-192-168-",
+    REGULAR_IP_OCTET,
+    "-",
+    REGULAR_IP_OCTET,
+    "\\b)",
+    REGULAR_PORTS
+);
+
+function scan(helper,msg) {
+
+    var ipRegex = new RegExp(patternPre.join(""),"gim");
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+
+    var counter = 0;
+    var message = String(msg.getReadablePayload());
+
+    if((matches = message.match(ipRegex)) != null){
+
+        matches.forEach(function(evidence){
+            helper.newAlert()
+                .setRiskConfidence(RISK_LOW, CONFIDENCE_MEDIUM)
+                .setName("Private IP Disclosure via WebSocket (script)")
+                .setDescription("A private IP (such as 10.x.x.x, 172.x.x.x, 192.168.x.x)\
+ or an Amazon EC2 private hostname (for example, ip-10-0-56-78) has been found in the HTTP\
+ response body. This information might be helpful for further attacks targeting\
+ internal systems.")
+                .setParam(counter++)
```

Comment:
I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

Reply:
Fixed in 068de3dec